stdlib/Dates/test/io.jl

```diff
@@ -545,4 +554,28 @@ end
     @test Time("13:24", "II:MMp") == Time("13:24", "HH:MM")
 end
 
+@testset "Issue #10561, two-digit year parsing ambiguities" begin
+    error_dates = Dict()
+    expected_error_dates = Dict()
+
+    for test_year in [0, 1, 100, 101, 1000, 1001, 1900, 1901, 2000, 2001, 2100, 2101]
+        test_date = Date(test_year, 01, 01)
+        fmt_string = "yy"
+        output_date = Date(Dates.format(test_date, fmt_string), fmt_string)
+
+        # Year dates which cannot be recovered when formatted with ambiguous year format "yy"
+        if test_date != output_date
+            error_dates[test_date] = output_date
+        end
+
+        # Dates which we expect to be unrecoverable given current implementation, e.g. all years besides the years 0001-0100
+        if test_year >= 100
+            expected_error_dates[test_date] = Date(mod(test_year, 100), 01, 01)
+        end
+    end
+
+
+    @test expected_error_dates == error_dates
```

Comment:
It will be rather difficult to read the output of this test when it fails. It would be better just doing:
```julia
@test output_date == Date(test_year % 100)
```
inside the `for` loop

Reply:
@omus Nice, that cut about 90% of the bulk out of this PR and made it much clearer. Let me know if it tracks with your reasoning. :)